### ISSUE TITLE
Execution stops, unable to interrupt [Spyder 4]

### ISSUE THREAD

**Issue by OverLordGoldDragon** [spyder-ide/spyder#11344]
Keras-TensorFlow training stops some time after launch, and am unable to KeyboardInterrupt execution; unsure, but it may also have occurred in a context outside of TensorFlow. [Demo](https://www.dropbox.com/s/z7mvic529hej2xo/hang_demo.mp4?dl=0); it's out of frame, but I also click the red square after Ctrl + C to stop execution - it just turns gray, nothing happens. I tried waiting 5+ mins (not in this clip), also no help.

**Comment by ccordoba12** [comment]
Please try to run your code in the Jupyter notebook to see what happens there. If you get the same effect, then it's not Spyder fault.

**Comment by OverLordGoldDragon** [comment]
Reproduced it on a simple pandas operation:

```python
np.sum([df[col].map(str) for col in cols])
```
`df` has 300000 rows, 3 columns, all integers - so no memory problem. I don't use Jupyter, but if I find something reproducible I'll try it on it also - but I doubt it's a pandas problem here.

**Comment by ccordoba12** [comment]
@dalthviz, please try to reproduce this one.

**Comment by OverLordGoldDragon** [comment]
Here we go:

```python
import numpy  as np
import pandas as pd

df = pd.DataFrame(np.random.randint(0, 300000, (300000, 3)), columns=['A', 'B', 'C'])
combined = np.sum([df['A'].map(str), df['B'].map(str)])
```
... but this also occurs in Jupyter. Though I'm unsure it's a useful comparison, as I found Jupyter less reliable than Spyder in interrupts - so it may be even worse off. Nonetheless, no longer reproducible if changing the `randint` range to something smaller - so it might be a memory issue after all, though strange with 20+ GB RAM available; [demo](https://www.dropbox.com/s/xt96uf4ssr9di0o/hang_demo2.mp4?dl=0)

**Comment by ccordoba12** [comment]
Thanks for the reproducible example. It tried it on terminal IPython and it's not possible to interrupt it even there. So It's Spyder's fault.

I think you're doing things in a very inefficient way (I don't understand why you're trying to create a single string with all elements of that dataframe one after the other), and you need to find a better one.

**Comment by OverLordGoldDragon** [comment]
@ccordoba12 Not quite what's happening; the operations are row-wise, so at most 12 characters are being concatenated per row - that'd make it at most 3,600,000 characters in one column distributed among 300,000 rows.

... or that's what I _thought_ I was doing; totally forgot to specify `axis=0` - now it executes instantly. My bad

The TensorFlow hanging was also in a context of stacked CuDNNLSTMs, so might again be a memory / CUDA problem there - I'll keep an eye for any other examples, but so far, Spyder appears clean.
